[0m[[0minfo[0m] [0mRunning Lab3 testjsy/test03_dynscope.jsy[0m
[0m[[0mdebug[0m] [0mWaiting for threads to exit or System.exit to be called.[0m
[0m[[0mdebug[0m] [0m  Classpath:[0m
[0m[[0mdebug[0m] [0m	B:\PLAssignments\lab3\target\scala-2.10\classes[0m
[0m[[0mdebug[0m] [0m	B:\PLAssignments\lab3\lib\liblab3.jar[0m
[0m[[0mdebug[0m] [0m	B:\PLAssignments\lab3\lib\scalatest_2.10-2.0.jar[0m
[0m[[0mdebug[0m] [0m	C:\Users\sean.donohoe\.ivy2\cache\org.scala-lang\scala-library\jars\scala-library-2.10.3.jar[0m
[0m[[0mdebug[0m] [0mWaiting for thread run-main-0 to terminate.[0m
[0m[[31merror[0m] [0m(run-main-0) scala.MatchError: ConstDecl(x,N(1.0),ConstDecl(r,Binary(Plus,Call(Function(None,x,N(2.0)),N(3.0)),Var(x)),Binary(Seq,Print(Var(r)),ConstDecl(x,N(1.0),ConstDecl(g,Function(None,y,Var(x)),ConstDecl(h,Function(None,x,Call(Var(g),N(2.0))),Binary(Seq,Print(Call(Var(h),N(3.0))),ConstDecl(x,N(42.0),ConstDecl(plus,Function(None,x,Function(None,y,Binary(Plus,Var(x),Var(y)))),Binary(Seq,Print(Call(Call(Var(plus),N(3.0)),N(4.0))),ConstDecl(dyn,Function(None,d,Binary(Plus,Var(x),N(3.0))),ConstDecl(over,Function(None,z,ConstDecl(x,Var(z),Call(Var(dyn),Var(x)))),Print(Call(Var(over),N(2.0))))))))))))))) (of class jsy.lab3.ast$ConstDecl)[0m
scala.MatchError: ConstDecl(x,N(1.0),ConstDecl(r,Binary(Plus,Call(Function(None,x,N(2.0)),N(3.0)),Var(x)),Binary(Seq,Print(Var(r)),ConstDecl(x,N(1.0),ConstDecl(g,Function(None,y,Var(x)),ConstDecl(h,Function(None,x,Call(Var(g),N(2.0))),Binary(Seq,Print(Call(Var(h),N(3.0))),ConstDecl(x,N(42.0),ConstDecl(plus,Function(None,x,Function(None,y,Binary(Plus,Var(x),Var(y)))),Binary(Seq,Print(Call(Call(Var(plus),N(3.0)),N(4.0))),ConstDecl(dyn,Function(None,d,Binary(Plus,Var(x),N(3.0))),ConstDecl(over,Function(None,z,ConstDecl(x,Var(z),Call(Var(dyn),Var(x)))),Print(Call(Var(over),N(2.0))))))))))))))) (of class jsy.lab3.ast$ConstDecl)
	at Lab3$.step(Lab3.scala:236)
	at Lab3$.loop$1(Lab3.scala:282)
	at Lab3$.iterateStep(Lab3.scala:288)
	at Lab3$$anonfun$processFile$2.apply$mcV$sp(Lab3.scala:319)
	at Lab3$$anonfun$processFile$2.apply(Lab3.scala:318)
	at Lab3$$anonfun$processFile$2.apply(Lab3.scala:318)
	at jsy.util.JsyApplication.handle(JsyApplication.scala:21)
	at Lab3$.processFile(Lab3.scala:318)
	at jsy.util.JsyApplication.jsy$util$JsyApplication$$doFile$1(JsyApplication.scala:39)
	at jsy.util.JsyApplication.main(JsyApplication.scala:50)
	at Lab3.main(Lab3.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at sbt.Run.invokeMain(Run.scala:67)
	at sbt.Run.run0(Run.scala:61)
	at sbt.Run.sbt$Run$$execute$1(Run.scala:51)
	at sbt.Run$$anonfun$run$1.apply$mcV$sp(Run.scala:55)
	at sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	at sbt.Run$$anonfun$run$1.apply(Run.scala:55)
	at sbt.Logger$$anon$4.apply(Logger.scala:85)
	at sbt.TrapExit$App.run(TrapExit.scala:248)
	at java.lang.Thread.run(Thread.java:745)
[0m[[0mdebug[0m] [0m	Thread run-main-0 exited.[0m
[0m[[0mdebug[0m] [0mInterrupting remaining threads (should be all daemons).[0m
[0m[[0mdebug[0m] [0mSandboxed run complete..[0m
java.lang.RuntimeException: Nonzero exit code: 1
	at scala.sys.package$.error(package.scala:27)
	at sbt.BuildCommon$$anonfun$toError$1.apply(Defaults.scala:1874)
	at sbt.BuildCommon$$anonfun$toError$1.apply(Defaults.scala:1874)
	at scala.Option.foreach(Option.scala:236)
	at sbt.BuildCommon$class.toError(Defaults.scala:1874)
	at sbt.Defaults$.toError(Defaults.scala:35)
	at sbt.Defaults$$anonfun$runTask$1$$anonfun$apply$36$$anonfun$apply$37.apply(Defaults.scala:707)
	at sbt.Defaults$$anonfun$runTask$1$$anonfun$apply$36$$anonfun$apply$37.apply(Defaults.scala:705)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
[0m[[31merror[0m] [0m(lab3/compile:[31mrun[0m) Nonzero exit code: 1[0m
